salt/cloud/clouds/vmware.py

```diff
@@ -3294,6 +3294,28 @@ def create_snapshot(name, kwargs=None, call=None):
                                                             vm_ref.snapshot.currentSnapshot)}
 
 
+def _find_snapshot_by_name(vm_ref, name):
```

Comment:
Can you split this function into two and restructure it as following to make it more clear for people:
```
def _get_snapshot_ref_helper(base_snapshot, snapshot_name):
    if base_snapshot.name == snapshot_name:
        return base_snapshot
    else:
        for snapshot in base_snapshot.childSnapshotList:
            snapshot_ref = _get_snapshot_ref_helper(snapshot, snapshot_name)
            if snapshot_ref is not None:
                return snapshot_ref

def _get_snapshot_ref_by_name(vm_ref, snapshot_name):
    snapshot_ref = None
    try:
        for root_snapshot in vm_ref.snapshot.rootSnapshotList:
            snapshot_ref = _get_snapshot_ref_helper(root_snapshot, snapshot_name)
            if snapshot_ref is not None:
                break
    except (IndexError, AttributeError):
        snapshot_ref = None

    return snapshot_ref
```